lldb/test/API/tools/lldb-dap/stepInTargets/TestDAP_stepInTargets.py

```diff
@@ -55,14 +55,23 @@ def test_basic(self):
         self.assertEqual(len(step_in_targets), 3, "expect 3 step in targets")
 
         # Verify the target names are correct.
-        self.assertEqual(step_in_targets[0]["label"], "bar()", "expect bar()")
-        self.assertEqual(step_in_targets[1]["label"], "bar2()", "expect bar2()")
-        self.assertEqual(
-            step_in_targets[2]["label"], "foo(int, int)", "expect foo(int, int)"
-        )
+        # The order of funcA and funcB may change depending on the compiler ABI.
+        funcA_target = None
+        funcB_target = None
+        for target in step_in_targets[0:2]:
+            if "funcB" in target["label"]:
+                funcB_target = target
+            elif "funcA" in target["label"]:
+                funcA_target = target
+            else:
+                self.fail(f"Unexpected step in target: {target}")
+
+        self.assertIsNotNone(funcA_target, "expect funcA")
+        self.assertIsNotNone(funcB_target, "expect funcB")
+        self.assertIn("foo", step_in_targets[2]["label"], "expect foo")
 
-        # Choose to step into second target and verify that we are in bar2()
-        self.stepIn(threadId=tid, targetId=step_in_targets[1]["id"], waitForStop=True)
+        # Choose to step into second target and verify that we are in funcB()
```

Comment:
funcB is not necessarily going to be the second target, so this comment is incorrect. However, instead of changing the comment, I think it'd be actually better to change the code so that it steps into the second function, regardless of what that function might be. This is because we would end up in the first function even if we did a "normal" step in operation rather than the fancy targetted one. Stepping into the second function verifies that we can correctly ignore the first function that gets called. (I wasn't the person who wrote the test, but I suspect this is roughly the motivation for it being written the way it is).

So basically, you could keep the `targetId=step_in_targets[1]["id"]` thing in the line below, but then change the expectation on line 77 to expect the function that we tried stepping into.

Reply:
Thanks for the explanation! This sounds like the right thing to do, however I believe I'm hitting a bug: the test's DAP always steps into funcB regardless of the option passed to `targetId`.  Would you be able to reproduce this? If it reproduces for you, I wonder if this has always been broken, given that IIRC there's no Windows x64 CI and this test only runs when the architecture is x86_64.

Reply:
I don't have access to a windows machine right now, but I can totally believe it that this has never worked there. It's quite possible that there's some platform-specific bit needed for this to work, which simply hasn't been implemented for windows yet. I don't really know what that could be though, as I don't know how this feature works.

Is this something you'd be willing to look into? If yes, you can try debugging lldb to see where things start to break down. Otherwise, we can just xfail the test and move on.

Reply:
I do not think anyone will have time to get a Windows machine, investigate and fix there. If there is indeed a bug on Windows, the easiest way is skip this test for Windows (with a comment mentioning it is not working there).

Reply:
That's reasonable. I'll mark as disabled for now and file a bug to investigate. I can look into it once I get the time.

Reply:
Updated in the latest commit.

Reply:
I used @expectFailureAll rather than @skipIf, not realizing that the test _must_ fail if the latter is used. Updated in the latest commit. I'll need help landing as I don't have write access. Thank you all for the reviews!